File: 45. Jump Game II/ac.cpp
#include<vector>
using namespace std;


class Solution {
public:
    int jump(vector<int>& nums) {
        int dp[nums.size() + 10];
        dp[0] = 0;
        int start = 0;
        int step;
        for (int i = 0; i < nums.size() && start + 1 < nums.size(); ++i)
        {
            step = i + nums[i];
            for (int j = start + 1; j <= step; ++j){
                dp[j] = dp[i] + 1;
            }
            start = step > start ? step : start;
            
        }
        return dp[nums.size() - 1];
    }
};

int main(){
    Solution s;
    vector<int> vec = {2, 1};
    s.jump(vec);
}

File: 334.递增三元子序列/ac.py
class Solution(object):
	def increasingTriplet(self, nums):
		"""
		:type nums: List[int]
		:rtype: bool
		"""
		a, b = float('inf'), float('inf')
		for n in nums:
			if n < a: a = n
			elif n > a and n < b:
				b = n
			elif n > b:
				return True

		return False
            

s = Solution()
print(s.increasingTriplet([1, 2, 0, 4]))